src/core/data/allocator.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "core/data/allocator.h"
+
+namespace legate {
+
+IntraTaskAllocator::IntraTaskAllocator(Legion::Memory::Kind kind) : target_kind_(kind) {}
+
+IntraTaskAllocator::~IntraTaskAllocator()
+{
+  for (auto& pair : buffers_) { pair.second.destroy(); }
+  buffers_.clear();
+}
+
+char* IntraTaskAllocator::allocate(size_t bytes)
+{
+  if (bytes == 0) return nullptr;
+
+  // Use 16-byte alignment
+  bytes = (bytes + 15) / 16 * 16;
```

Comment:
Let's make the alignment configurable. We can take this as a constructor argument.

Also, let's pass the alignment to `DeferredBuffer`'s constructor: https://github.com/StanfordLegion/legion/blob/stable/runtime/legion.h#L3223, instead of computing the aligned size manually here. You would need to change `create_buffer` to take the alignment.

Reply:
16byte seems to be default for DeferredBuffers anyway - so this manual adjustment is not needed. Do we want to define a default in create_buffer as well or do we want additional constructors with the alignment parameter?

Reply:
You can make the argument optional in both the allocator's constructor and `create_buffer`, with the default value being 16.